Content.Server/StationEvents/Events/RandomFundingRule.cs

```diff
@@ -0,0 +1,82 @@
+using Content.Server.StationEvents.Components;
+using Content.Server.Cargo.Systems;
+using Content.Shared.Dataset;
+using Content.Shared.GameTicking.Components;
+using Content.Shared.Random.Helpers;
+using Content.Shared.Cargo.Components;
+using Robust.Shared.Random;
+using Robust.Shared.Prototypes;
+
+using System.Linq;
+
+namespace Content.Server.StationEvents.Events;
+
+public sealed class RandomFundingRule : StationEventSystem<RandomFundingRuleComponent>
+{
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly CargoSystem _cargo = default!;
+    [Dependency] private readonly IPrototypeManager _prototype = default!;
+
+
+    protected override void Started(EntityUid uid, RandomFundingRuleComponent component, GameRuleComponent gameRule, GameRuleStartedEvent args)
+    {
+        base.Started(uid, component, gameRule, args); //does all the pre-amble in the base StationEventSystem
+
+        if (!TryGetRandomStation(out var station)) //in theory the event-eligible target should always be the station, which has a bank account...
+            return;
+
+        // No station to deduct from.
+        if (TryComp(station, out StationBankAccountComponent? bank))//...but if no bank account found cancel immediately.
+        {
+            var payment = (int)(component.BaseCash * (_random.Next(component.MaxMult) + 1)); //component-defined base cash multiplied by random number between 1 and component-defined maximum multiplier.
+            var dep = "test"; //department suffix used for announcement, should always be replaced
```

Comment:
rather than set it to a placeholder
ensure it always gets set
```suggestion
            string dep;
```

Reply:
✓